internal/clients/tfclient/tfclient.go

```diff
@@ -135,7 +136,7 @@ func TerraformSetupBuilderNoTracking(version, providerSource, providerVersion st
 
 		var userCredential btp.UserCredential
 		if err := json.Unmarshal(ServiceAccountSecretData, &userCredential); err != nil {
-			return ps, err
+			return ps, errors.Wrap(err, errCouldNotParseUserCredential)
```

Comment:
Just seeing that now sorry. 
There is actually another function called `TerraformSetupBuilder` above which does pretty much the same thing. 
Probably makes sense to combined those, but for now lets just add the same error message on parsing there like here.

I will create a follow up issue for adding test cases for that.